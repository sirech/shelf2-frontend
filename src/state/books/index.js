import * as constants from './constants'
import * as actions from './actions'

export { actions, constants }
export { default as reducer } from './reducers'
