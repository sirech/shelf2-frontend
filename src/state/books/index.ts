import * as constants from './constants'
import * as actions from './slice'

export { actions, constants }
export { default as booksSelector } from './selectors'
export { default as reducer } from './slice'
