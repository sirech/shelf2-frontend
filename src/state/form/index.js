import * as constants from './constants'
import * as actions from './actions'

export { constants, actions }
export { default as formSelector, bookSelector } from './selectors'
export { default as reducer } from './reducers'

export { modelName } from './utils'
