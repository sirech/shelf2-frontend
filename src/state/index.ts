import { ThunkAction, UnknownAction, configureStore } from '@reduxjs/toolkit'

import { constants as booksConstants, reducer as booksReducer } from './books'
import {
  constants as searchConstants,
  reducer as searchReducer,
} from './search'
import { constants as yearsConstants, reducer as yearsReducer } from './years'
import { constants as loginConstants, reducer as loginReducer } from './login'

const store = configureStore({
  reducer: {
    [booksConstants.namespace]: booksReducer,
    [searchConstants.namespace]: searchReducer,
    [yearsConstants.namespace]: yearsReducer,
    [loginConstants.namespace]: loginReducer,
  },
})

export type RootState = ReturnType<typeof store.getState>
export type AppDispatch = typeof store.dispatch
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  UnknownAction
>

export default store
