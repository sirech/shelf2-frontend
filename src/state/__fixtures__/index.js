import R from 'ramda'

import books from './books'
import years from './years'
import form from './form'
import login from './login'

const modal = () => ({
  opened: false
})

const search = R.pipe(books, R.omit(['activeYear']))

const state = () => {
  return {
    books: books(),
    search: search(),
    years: years(),
    form: form(),
    modal: modal(),
    login: login()
  }
}

export { books, search, years, form, login }
export default state
