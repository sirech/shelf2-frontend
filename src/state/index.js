import { combineReducers } from 'redux'

import * as R from 'ramda'

import { constants as booksConstants, reducer as booksReducer } from './books'
import {
  constants as searchConstants,
  reducer as searchReducer,
} from './search'
import { constants as yearsConstants, reducer as yearsReducer } from './years'
import { constants as formConstants, reducer as formReducer } from './form'
import { constants as modalConstants, reducer as modalReducer } from './modal'
import { constants as loginConstants, reducer as loginReducer } from './login'

export const actionPicker = list => {
  return actions => {
    const h = R.pick(list)(actions)

    if (Object.keys(h).length !== list.length) {
      throw new Error(
        `Not all requested actions (${list}) exist in the given actions`
      )
    }

    return h
  }
}

const rootReducer = combineReducers({
  [booksConstants.namespace]: booksReducer,
  [searchConstants.namespace]: searchReducer,
  [yearsConstants.namespace]: yearsReducer,
  [formConstants.namespace]: formReducer,
  [modalConstants.namespace]: modalReducer,
  [loginConstants.namespace]: loginReducer,
})

export default rootReducer
