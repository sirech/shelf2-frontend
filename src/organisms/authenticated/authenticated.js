// @flow

import React from 'react'
import { connect } from 'react-redux'

import * as R from 'ramda'
import wrapDisplayName from 'recompose/wrapDisplayName'
import { createStructuredSelector } from 'reselect'

import { authenticatedSelector } from 'state/login'

type Props = {
  authenticated: boolean,
}

export default (shouldBeAuthenticated: boolean = true) => (
  Component: Class<React$Component<*, *>>
) => {
  class AuthenticatedComponent extends React.Component {
    props: Props

    conditionalRender() {
      if (this.props.authenticated === shouldBeAuthenticated) {
        return <Component {...R.omit(['authenticated'], this.props)} />
      }

      return null
    }

    render() {
      return this.conditionalRender()
    }
  }

  AuthenticatedComponent.displayName = wrapDisplayName(
    Component,
    'authenticatedComponent'
  )

  return connect((state, props) =>
    createStructuredSelector({
      authenticated: authenticatedSelector,
    })(state)
  )(AuthenticatedComponent)
}
