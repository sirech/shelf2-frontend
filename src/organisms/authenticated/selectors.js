import R from 'ramda'

import { loginSelector } from '../../state/login'

const selector = R.pipe(loginSelector, R.path(['authenticated']))
export default selector
