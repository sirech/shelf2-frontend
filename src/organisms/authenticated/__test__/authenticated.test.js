import React from 'react'

import { fullRender } from 'test'
import state from 'state/__fixtures__'

import authenticated from '../authenticated'

describe('authenticated', () => {
  // eslint-disable-next-line
  const Dummy = () => <div>Mock</div>

  it('renders the component if authenticated is required', () => {
    const Component = authenticated()(Dummy)
    const { component } = fullRender(<Component />, state())
    expect(component.toJSON()).toMatchSnapshot()
  })

  it('does not render the component if authenticated is required and authentincation is not set', () => {
    const currentState = state()
    currentState.login.authenticated = false
    const Component = authenticated()(Dummy)
    const { component } = fullRender(<Component />, currentState)
    expect(component.toJSON()).toBeNull()
  })

  it('does not render the component if authenticated is not required', () => {
    const Component = authenticated(false)(Dummy)
    const { component } = fullRender(<Component />, state())
    expect(component.toJSON()).toBeNull()
  })
})
