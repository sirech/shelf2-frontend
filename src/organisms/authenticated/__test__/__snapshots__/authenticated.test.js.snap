// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`authenticated does not render the component if authenticated is not required 1`] = `null`;

exports[`authenticated renders the component if authenticated is required 1`] = `
<div>
  Mock
  stuff
</div>
`;
